core/server/master/src/test/java/alluxio/master/file/FileSystemMasterSyncMetadataConcurrentTest.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file;
+
+import static org.junit.Assert.assertEquals;
+
+import alluxio.AlluxioURI;
+import alluxio.collections.Pair;
+import alluxio.concurrent.jsr.CompletableFuture;
+import alluxio.exception.AccessControlException;
+import alluxio.exception.InvalidPathException;
+import alluxio.file.options.DescendantType;
+import alluxio.grpc.FileSystemMasterCommonPOptions;
+import alluxio.master.file.meta.InodeTree;
+import alluxio.master.file.meta.LockingScheme;
+import alluxio.underfs.UnderFileSystem;
+
+import com.google.common.math.IntMath;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({UnderFileSystem.Factory.class})
+public class FileSystemMasterSyncMetadataConcurrentTest
+    extends FileSystemMasterSyncMetadataTestBase {
+  private final int mNumDirsPerLevel = 2;
+  private final int mNumLevels = 3;
+  private final int mNumExpectedInodes = IntMath.pow(mNumDirsPerLevel, mNumLevels + 1) - 1;
+
+  @Override
+  public void before() throws Exception {
+    super.before();
+
+    createUFSHierarchy(0, mNumLevels, "", mNumDirsPerLevel);
+
+    mUfs.mIsSlow = true;
+    mUfs.mSlowTimeMs = 500;
+    // verify the files don't exist in alluxio
+    assertEquals(1, mFileSystemMaster.getInodeTree().getInodeCount());
+  }
+
+  @Test
+  public void syncTheSameDirectory() throws Exception {
```

Comment:
can we add coverage for the loadMetadata case as well? I see that sync case is covered pretty well here. I am just worried that loadMetadata might hit a corner case?

Reply:
done